### DISCUSSION TITLE
How to remove taxonomy value from library field

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/pnpcore#527] [Q&A]
I am trying to  clear existing taxonomy selection with the SDK 
No idea where to start :-)
The field is single value managed metadata
Thanks

**Comment by jansenbe** [comment]
@Ofer-Gal : just set the value to null e.g. and than call one of the update methods

```csharp
myItem["TaxonomyField"] = null;
```

**Reply by Ofer-Gal** [reply]
This used to work but now,  1.7.0 maybe it does not clear the field as explained in https://pnp.github.io/pnpcore/using-the-sdk/listitems-fields.html#taxonomy-fields
// Or clear the taxonomy field
addedItem["MyField"] = null;

Any idea how to do it?

**Reply by jansenbe** [reply]
@Ofer-Gal : using the latest nightly build this works for me, tested with a regular update as with a system update:

```csharp
var list = await context.Web.Lists.GetByTitleAsync("FieldTypes", p=>p.Items);
var item = list.Items.AsRequested().First();

// Clear taxonomy values and update item
item["MMSingle"] = null;
(item["MMMultiple"] as IFieldValueCollection).Values.Clear();

// Test 1: Update using rest => OK
//await item.UpdateAsync();

// Test 2: Update using csom => OK
await item.SystemUpdateAsync();
```

**Reply by Ofer-Gal** [reply]
Only way I could make it work with version 1.7.0, was like so: The Field name is Document Type (internal DocumentType)
```
  IList DPCDocumentsList = await pnpContext.Web.Lists.GetByTitleAsync("DPC Documents", 
      l => l.Fields.QueryProperties(l => l.Id, l => l.InternalName, l => l.FieldTypeKind ,l=> l.Title)); //, p => p.Items
 var myf = DPCDocumentsList.Fields.First(t => t.Title == "Document Type_0");  //Hidden !!!
```

Then for each item:
```
                  listItem[myf.InternalName] = null;       // Clear the DocumentType
                  listItem["DPCNotes"] = "";       // Clear the DPCNotes
                  await listItem.SystemUpdateAsync();
```
Maybe add this option to the documentation?
I think the API are changing for each tenant every once in a while 

Thank you
